utilities/backupable/backupable_db.cc

```diff
@@ -46,6 +46,15 @@
 
 namespace ROCKSDB_NAMESPACE {
 
+namespace {
+inline bool IsSstFile(const std::string& fname) {
+  return fname.length() > 4 && fname.rfind(".sst") == fname.length() - 4;
+}
+inline uint32_t ChecksumStrToInt32(const std::string& checksum_str) {
```

Comment:
Although this is compatible with the existing way checksums are put into the file name, this does not generalize to arbitrary custom checksums (which I expect we will be doing next). Maybe use hexadecimal encoding of the checksum bytes?

Reply:
Right. But I'm a bit worried about using hexadecimal encoding for the default the checksums as they are stored in the meta files. 

1. We could add some logic in the code to distinguish between unit32 values and hex values we read from old/new meta files. Not sure whether it is a preferable.

2. Maybe we only use hexadecimal encoding in the filename but keep unit32 values in the meta files. Not sure whether it will be confusing to users.

What are your thoughts? :)

Reply:
Good points. We certainly want to keep forward compatibility with this change, so we have to use decimal in the backup meta.

With hex in file names, there is potential for awkwardness for users who look at those file names with the old scheme and with the new scheme. (Btw, I think that's minor compared to breaking forward compatibility with even an opt-in backup manifest schema change.) But while making a change to the file naming scheme, I think we should work toward a scheme that makes sense well into the future without being burdened too much by the past. Hopefully the old scheme will be forgotten by users and only exist as a relic in the code for backward compatibility.

Peering into that future, does it make sense to encode custom file checksums, which we pass around as byte strings, into file names in decimal? No, decimal is terrible for byte strings of unknown length. Should GetFileChecksumGenCrc32cFactory be encoded differently from other custom file checksums? That seems weird to me if we make our story "your hex-encoded checksum from the db manifest, when available, will be used in the backup file name." And I think that's the best long-term story. I'd be happy if this PR didn't have the Crc32c limitation to better match that story ASAP, but I haven't considered all the potential difficulties.